x-pack/plugins/threat_intelligence/cypress/e2e/cases.cy.ts

```diff
@@ -0,0 +1,137 @@
+const CASES = 'app/security/cases';
+
+const createNewCaseFromCases = () => {
+  cy.visit(CASES);
+  cy.get(CREAT_CASE_BUTTON).click();
+  cy.get(NEW_CASE_NAME_INPUT).type('case');
+  cy.get(NEW_CASE_DESCRIPTION_INPUT).type('case description');
+  cy.get(NEW_CASE_CREATE_BUTTON).click();
+};
+
+const createNewCaseFromTI = () => {
+  cy.get(NEW_CASE_NAME_INPUT).type('case');
+  cy.get(NEW_CASE_DESCRIPTION_INPUT).type('case description');
+  cy.get(NEW_CASE_CREATE_BUTTON).click();
+};
+
+const selectExistingCase = () => {
+  cy.wait(500); // TODO find a better way to wait for the table to render
+  cy.get(SELECT_EXISTING_CASE).should('exist').contains('Select').click();
+};
+
+const navigateToNewCaseAndCheckAddedComment = () => {
+  cy.get(VIEW_CASE_TOASTER_LINK).click();
+  cy.get(CASE_COMMENT_EXTERNAL_REFERENCE)
+    .should('exist')
+    .and('contain.text', 'added an indicator of compromise')
+    .and('contain.text', 'Indicator name')
+    .and('contain.text', 'Indicator type')
+    .and('contain.text', 'Feed name');
+};
+
+const deleteCase = () => {
+  cy.get(CASE_ACTION_WRAPPER).find(CASE_ELLIPSE_BUTTON).click();
+  cy.get(CASE_ELLIPSE_DELETE_CASE_OPTION).click();
+  cy.get(CASE_ELLIPSE_DELETE_CASE_CONFIRMATION_BUTTON).click();
+};
+
+describe('Indicators', () => {
+  before(() => {
+    esArchiverLoad('threat_intelligence/indicators_data');
+  });
+  after(() => {
+    esArchiverUnload('threat_intelligence/indicators_data');
+  });
+
+  describe('Indicators cases interactions', () => {
+    it('should add to existing case when clicking on the button in the indicators table', () => {
+      createNewCaseFromCases();
+
+      cy.visit(THREAT_INTELLIGENCE);
+      selectRange();
+
+      cy.get(INDICATORS_TABLE_MORE_ACTION_BUTTON_ICON).first().click();
+      cy.get(INDICATORS_TABLE_ADD_TO_EXISTING_CASE_BUTTON_ICON).first().click();
+
+      selectExistingCase();
+      navigateToNewCaseAndCheckAddedComment();
+      deleteCase();
+    });
+
+    it('should add to new case when clicking on the button in the indicators table', () => {
+      cy.visit(THREAT_INTELLIGENCE);
+      selectRange();
+
+      cy.get(INDICATORS_TABLE_MORE_ACTION_BUTTON_ICON).first().click();
+      cy.get(INDICATORS_TABLE_ADD_TO_NEW_CASE_BUTTON_ICON).first().click();
+      createNewCaseFromTI();
+
+      navigateToNewCaseAndCheckAddedComment();
+      deleteCase();
+    });
+
+    it('should add to existing case when clicking on the button in the indicators flyout', () => {
+      createNewCaseFromCases();
+
+      cy.visit(THREAT_INTELLIGENCE);
+      selectRange();
+
+      cy.get(TOGGLE_FLYOUT_BUTTON).first().click({ force: true });
+      cy.get(FLYOUT_TAKE_ACTION_BUTTON).first().click();
+      cy.get(FLYOUT_ADD_TO_EXISTING_CASE_ITEM).first().click();
+
+      selectExistingCase();
+      navigateToNewCaseAndCheckAddedComment();
+      deleteCase();
+    });
+
+    it('should add to new case when clicking on the button in the indicators flyout', () => {
```

Comment:
are we sure we want to hide if only the name is absent?

Also to make the UI perfect this would be a bit tricker, as now there aren't any options in the 3-dot menu context menu int he table, which means we should probably hide the button entirely, which means the column is going to be way too wide for 2 buttons...